packages/messenger-widget/src/hooks/messages/acknowledge/AcklowledgementManager.ts

```diff
@@ -0,0 +1,90 @@
+import { Account, ProfileKeys } from '@dm3-org/dm3-lib-profile';
+import { MessageModel } from '../useMessage';
+import {
+    createReadOpenMessage,
+    createReadReceiveMessage,
+} from '@dm3-org/dm3-lib-messaging';
+import { ContactPreview } from '../../../interfaces/utils';
+
+//Class that takes care of acknowledging messages
+export class AcknowledgmentManager {
+    private readonly account: Account;
+    private readonly profileKeys: ProfileKeys;
+    private readonly addMessage: Function;
+    constructor(
+        account: Account,
+        profileKeys: ProfileKeys,
+        addMessage: Function,
+    ) {
+        this.account = account;
+        this.profileKeys = profileKeys;
+        this.addMessage = addMessage;
+    }
+    private ackOpened = async (
+        contactAliasName: string,
+        messageModel: MessageModel,
+    ) => {
+        const readedMsg = await createReadOpenMessage(
```

Comment:
read is a strong verb: to read / read / read
-> rename to readMsg or readMessages or messagesRead